src/Serializers/SnakSerializer.php

```diff
@@ -23,10 +24,17 @@ class SnakSerializer implements DispatchableSerializer {
 	private $dataValueSerializer;
 
 	/**
+	 * @var bool
+	 */
+	private $serializeSnaksWithHash;
+
+	/**
 	 * @param Serializer $dataValueSerializer
+	 * @param bool $serializeSnaksWithHash
 	 */
-	public function __construct( Serializer $dataValueSerializer ) {
+	public function __construct( Serializer $dataValueSerializer, $serializeSnaksWithHash = true ) {
```

Comment:
Think that name is not 100% correct grammar wise. I'd go for either `serializeWithHash` or something like `includeHash`